feat(uu): add up-cli, update deps

haarchri opened the pull request:
<!--
Thank you for helping to improve Upbound!

Please read through https://git.io/fj2m9 if this is your first time opening an
Upbound pull request. Find us in https://slack.crossplane.io/messages/upbound if
you need any help contributing.
-->

### Description of your changes
- add up-cli for argocd https://docs.upbound.io/mcp/gitops/#add-the-up-cli-init-container-to-argo
- update deps to fit platform-ref-aws again
- update repo structure
- update for make render
- update for make yamllint
- change submodule to crossplane/build

<!--
Briefly describe what this pull request does. Be sure to direct your reviewers'
attention to anything that needs special consideration.

We love pull requests that resolve an open issue. If yours does, use the below
line to indicate which issue your PR fixes, for example "Fixes #500":
-->

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested
make e2e
<!--
Before reviewers can be confident in the correctness of this pull request, it
needs to tested and shown to be correct. Briefly describe the testing that has
already been done or which is planned for this change. Consider pasting snippets
with the commands or configurations you used to test, in case you or a reviewer
needs to repeat the test in future.
-->

haarchri commented:
/test-examples

kaessert commented on examples/argocd-xr.yaml:
```diff
@@ -12,32 +15,35 @@ spec:
-        - Composition
-        clusters:
-        - "ctp-*"
+    # resourceExclusions: |
```
Small comment would be nice why it's commented out :)

ytsarev commented on apis/composition.yaml:
```diff
@@ -211,6 +242,34 @@ spec:
+                            - sh
+                            - -c
+                          args:
+                            - wget -qO /plugin/up https://cli.upbound.io/stable/current/bin/${OS}_${ARCH}/up && chmod +x /plugin/up
```
```suggestion
                            - wget -qO /plugin/up https://cli.upbound.io/stable/current/bin/${OS}_${ARCH}/up && chmod +x /plugin/up
```
can we run something like `$(uname -m)` or `$(arch)` to dynamically determine the architecture from within container?

ytsarev commented on apis/definition.yaml:
```diff
@@ -71,8 +72,13 @@ spec:
+                              default: amd64
                           required:
                             - version
+                            - arch
```
Exposing underlying architecture to consumers looks like a serious abstraction leak to me. Please see my comment above to use `$(uname -m)` or `$(arch)` commands to dynamically resolve the current runtime architecture

haarchri commented:
the make e2e target is working locally - but the shared workflow in master is using an old cache version - so i wonder if we want to merge and use the renovate job to make this up2date?

```
make e2e       
13:33:30 [ .. ] installing up v0.35.0
13:33:36 [ OK ] installing up v0.35.0
13:33:37 [ .. ] Building package configuration-gitops-argocd-v0.10.0-dirty.xpkg for linux_arm64
xpkg saved to /private/tmp/configuration-gitops-argocd/_output/xpkg/linux_arm64/configuration-gitops-argocd-v0.10.0-dirty.xpkg
13:33:38 [ OK ] Built package configuration-gitops-argocd-v0.10.0-dirty.xpkg for linux_arm64
13:33:38 [ .. ] installing kind v0.23.0
13:33:39 [ OK ] installing kind v0.23.0
13:33:39 [ .. ] deleting controlplane
Deleting cluster "uptest-configuration-gitops-argocd" ...
Deleted nodes: ["uptest-configuration-gitops-argocd-control-plane"]
13:33:41 [ OK ] deleting controlplane
13:33:41 [ .. ] installing helm v3.9.1
13:33:49 [ OK ] installing helm v3.9.1
13:33:49 [ .. ] setting up controlplane
Creating cluster "uptest-configuration-gitops-argocd" ...
 ✓ Ensuring node image (kindest/node:v1.30.0) 🖼
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
Set kubectl context to "kind-uptest-configuration-gitops-argocd"
You can now use your cluster with:

kubectl cluster-info --context kind-uptest-configuration-gitops-argocd

Not sure what to do next? 😅  Check out https://kind.sigs.k8s.io/docs/user/quick-start/
13:34:05 [ .. ] setting kubectl context to kind-uptest-configuration-gitops-argocd
Switched to context "kind-uptest-configuration-gitops-argocd".
"crossplane-build-module" has been added to your repositories
Hang tight while we grab the latest from your chart repositories...
Update Complete. ⎈Happy Helming!⎈
13:34:09 [ .. ] setting up crossplane core with args --enable-usages
NAME: crossplane
LAST DEPLOYED: Wed Mar  5 13:34:09 2025
NAMESPACE: upbound-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
By proceeding, you are accepting to comply with terms and conditions in https://licenses.upbound.io/upbound-software-license.html

✨ Thank you for installing Universal Crossplane!
🚀 Install packages from https://marketplace.upbound.io to get started!
13:34:10 [ .. ] patching Crossplane with dev sidecar
deployment.apps/crossplane patched
deployment.apps/crossplane condition met
pod/crossplane-6ff649df6-2gm7j condition met
13:34:34 [ OK ] patching Crossplane with dev sidecar
13:34:34 [ .. ] copying local xpkg cache to Crossplane pod
xpkg contents extracted to /private/tmp/configuration-gitops-argocd/_output/xpkg/cache/configuration-gitops-argocd-v0.10.0-dirty.gz
13:34:34 [ OK ] copying local xpkg cache to Crossplane pod
13:34:34 [ .. ] deploying configuration package configuration-gitops-argocd v0.10.0-dirty
configuration.pkg.crossplane.io/configuration-gitops-argocd created
13:34:35 [ OK ] deploying configuration package configuration-gitops-argocd v0.10.0-dirty
13:34:35 [ .. ] running automated tests
SKIP_DEPLOY_ARGO= KUBECTL=/private/tmp/configuration-gitops-argocd/.cache/tools/darwin_arm64/kubectl-v1.24.3 CHAINSAW=/private/tmp/configuration-gitops-argocd/.cache/tools/darwin_arm64/chainsaw-0.2.0 CROSSPLANE_CLI=/private/tmp/configuration-gitops-argocd/.cache/tools/darwin_arm64/crossplane-cli-v1.18.0 CROSSPLANE_NAMESPACE=upbound-system YQ=/private/tmp/configuration-gitops-argocd/.cache/tools/darwin_arm64/yq-v4.40.5 /private/tmp/configuration-gitops-argocd/.cache/tools/darwin_arm64/uptest-v1.2.0 e2e examples/argocd-xr.yaml --data-source="" --setup-script=test/setup.sh  --skip-update --default-timeout=2400s || (echo `date +%H:%M:%S` [FAIL] && false)
2025/03/05 13:34:35 Skipping update step because the root resource does not exist
2025/03/05 13:34:35 Skipping update step because the skip-delete option is set to true
2025/03/05 13:34:35 Written test files: /var/folders/ym/wqrhc4wd0mj6b1wbkc25s0_80000gn/T/uptest-e2e
2025/03/05 13:34:35 Running chainsaw tests at /var/folders/ym/wqrhc4wd0mj6b1wbkc25s0_80000gn/T/uptest-e2e
2025/03/05 13:34:35 Version: 0.2.0
2025/03/05 13:34:35 Loading default configuration...
2025/03/05 13:34:35 - Using test file: 00-apply.yaml
2025/03/05 13:34:35 - TestDirs [/var/folders/ym/wqrhc4wd0mj6b1wbkc25s0_80000gn/T/uptest-e2e/case]
2025/03/05 13:34:35 - SkipDelete true
2025/03/05 13:34:35 - FailFast false
2025/03/05 13:34:35 - ReportFormat ''
2025/03/05 13:34:35 - ReportName 'chainsaw-report'
2025/03/05 13:34:35 - Namespace ''
2025/03/05 13:34:35 - FullName false
2025/03/05 13:34:35 - IncludeTestRegex ''
2025/03/05 13:34:35 - ExcludeTestRegex ''
2025/03/05 13:34:35 - ApplyTimeout 5s
2025/03/05 13:34:35 - AssertTimeout 30s
2025/03/05 13:34:35 - CleanupTimeout 30s
2025/03/05 13:34:35 - DeleteTimeout 15s
2025/03/05 13:34:35 - ErrorTimeout 30s
2025/03/05 13:34:35 - ExecTimeout 5s
2025/03/05 13:34:35 - Parallel 1
2025/03/05 13:34:35 - NoCluster false
2025/03/05 13:34:35 Loading tests...
2025/03/05 13:34:35 - apply (/var/folders/ym/wqrhc4wd0mj6b1wbkc25s0_80000gn/T/uptest-e2e/case)
2025/03/05 13:34:35 Loading values...
2025/03/05 13:34:35 Running tests...
2025/03/05 13:34:35 === RUN   chainsaw
2025/03/05 13:34:35 === PAUSE chainsaw
2025/03/05 13:34:35 === CONT  chainsaw
2025/03/05 13:34:35 === RUN   chainsaw/apply
2025/03/05 13:34:35 === PAUSE chainsaw/apply
2025/03/05 13:34:35 === CONT  chainsaw/apply
2025/03/05 13:34:35     | 13:34:35 | apply | @setup                   | CREATE    | OK    | v1/Namespace @ chainsaw-one-whale
2025/03/05 13:34:35     | 13:34:35 | apply | Run Setup Script         | TRY       | RUN   |
2025/03/05 13:34:35     | 13:34:35 | apply | Run Setup Script         | CMD       | RUN   |
2025/03/05 13:34:35         === COMMAND
2025/03/05 13:34:35         /tmp/configuration-gitops-argocd/test/setup.sh
2025/03/05 13:34:54     | 13:34:54 | apply | Run Setup Script         | CMD       | LOG   |
2025/03/05 13:34:54         === STDOUT
2025/03/05 13:34:54         Running setup.sh
2025/03/05 13:34:54         Waiting until all configurations are healthy/installed...
2025/03/05 13:34:54         configuration.pkg.crossplane.io/configuration-gitops-argocd condition met
2025/03/05 13:34:54         configuration.pkg.crossplane.io/configuration-gitops-argocd condition met
2025/03/05 13:34:54         Waiting until all installed provider packages are healthy...
2025/03/05 13:34:54         provider.pkg.crossplane.io/upbound-provider-helm condition met
2025/03/05 13:34:54         provider.pkg.crossplane.io/upbound-provider-kubernetes condition met
2025/03/05 13:34:54         Waiting for all pods to come online...
2025/03/05 13:34:54         deployment.apps/crossplane condition met
2025/03/05 13:34:54         deployment.apps/crossplane-contrib-function-patch-and-transform-6a1ab24d2512 condition met
2025/03/05 13:34:54         deployment.apps/crossplane-rbac-manager condition met
2025/03/05 13:34:54         deployment.apps/upbound-provider-helm-9fcd9ac09d15 condition met
2025/03/05 13:34:54         deployment.apps/upbound-provider-kubernetes-71953a1e5c15 condition met
2025/03/05 13:34:54         Waiting for all XRDs to be established...
2025/03/05 13:34:54         compositeresourcedefinition.apiextensions.crossplane.io/xargo.gitops.platform.upbound.io condition met
2025/03/05 13:34:54         Installing providerconfigs
2025/03/05 13:34:54         providerconfig.helm.crossplane.io/configuration-gitops-argocd created
2025/03/05 13:34:54         providerconfig.kubernetes.crossplane.io/configuration-gitops-argocd created
2025/03/05 13:34:54         Installed providerconfigs
2025/03/05 13:34:54         Adding provider-helm Service Account permissions
2025/03/05 13:34:54         clusterrolebinding.rbac.authorization.k8s.io/provider-helm-admin-binding created
2025/03/05 13:34:54         Added provider-helm Service Account permissions
2025/03/05 13:34:54         Adding provider-kubernetes Service Account permissions
2025/03/05 13:34:54         clusterrolebinding.rbac.authorization.k8s.io/provider-kubernetes-admin-binding created
2025/03/05 13:34:54         Added provider-kubernetes Service Account permissions
2025/03/05 13:34:54     | 13:34:54 | apply | Run Setup Script         | CMD       | DONE  |
2025/03/05 13:34:54     | 13:34:54 | apply | Run Setup Script         | TRY       | DONE  |
2025/03/05 13:34:54     | 13:34:54 | apply | Apply Resources          | TRY       | RUN   |
2025/03/05 13:34:54     | 13:34:54 | apply | Apply Resources          | APPLY     | RUN   | gitops.platform.upbound.io/v1alpha1/XArgo @ configuration-gitops-argocd
2025/03/05 13:34:54     | 13:34:54 | apply | Apply Resources          | CREATE    | OK    | gitops.platform.upbound.io/v1alpha1/XArgo @ configuration-gitops-argocd
2025/03/05 13:34:54     | 13:34:54 | apply | Apply Resources          | APPLY     | DONE  | gitops.platform.upbound.io/v1alpha1/XArgo @ configuration-gitops-argocd
2025/03/05 13:34:54     | 13:34:54 | apply | Apply Resources          | SCRIPT    | RUN   |
2025/03/05 13:34:54         === COMMAND
2025/03/05 13:34:54         /bin/sh -c echo "Runnning annotation script"
2025/03/05 13:34:54         ${KUBECTL} annotate xargo.gitops.platform.upbound.io/configuration-gitops-argocd upjet.upbound.io/test=true --overwrite
2025/03/05 13:34:54     | 13:34:54 | apply | Apply Resources          | SCRIPT    | LOG   |
2025/03/05 13:34:54         === STDOUT
2025/03/05 13:34:54         Runnning annotation script
2025/03/05 13:34:54         xargo.gitops.platform.upbound.io/configuration-gitops-argocd annotated
2025/03/05 13:34:54     | 13:34:54 | apply | Apply Resources          | SCRIPT    | DONE  |
2025/03/05 13:34:54     | 13:34:54 | apply | Apply Resources          | TRY       | DONE  |
2025/03/05 13:34:54     | 13:34:54 | apply | Assert Status Conditions | TRY       | RUN   |
2025/03/05 13:34:54     | 13:34:54 | apply | Assert Status Conditions | ASSERT    | RUN   | gitops.platform.upbound.io/v1alpha1/XArgo @ configuration-gitops-argocd
2025/03/05 13:35:06 crossplane trace logs 2025-03-05 13:35:06.216331 +0100 CET m=+30.758394835
NAME                                           RESOURCE             SYNCED   READY   STATUS
XArgo/configuration-gitops-argocd                                   True     False   Creating: Unready resources: argoApplicationSet, releaseArgo
├─ Release/configuration-gitops-argocd-hgtw4   releaseArgo          -        -       
└─ Object/configuration-gitops-argocd-n9t5f    argoApplicationSet   False    -       ReconcileError: observe failed: cannot get object: failed to get API group resources: unable to retrieve the complete list of server APIs: argoproj.io/v1alpha1: the server could not find the requested resource

2025/03/05 13:35:35 crossplane trace logs 2025-03-05 13:35:35.657393 +0100 CET m=+60.199540251
NAME                                           RESOURCE             SYNCED   READY   STATUS
XArgo/configuration-gitops-argocd                                   True     False   Creating: Unready resources: argoApplicationSet, releaseArgo
├─ Release/configuration-gitops-argocd-hgtw4   releaseArgo          True     True    Available
└─ Object/configuration-gitops-argocd-n9t5f    argoApplicationSet   False    -       ReconcileError: observe failed: cannot get object: failed to get API group resources: unable to retrieve the complete list of server APIs: argoproj.io/v1alpha1: the server could not find the requested resource

2025/03/05 13:35:53     | 13:35:53 | apply | Assert Status Conditions | ASSERT    | DONE  | gitops.platform.upbound.io/v1alpha1/XArgo @ configuration-gitops-argocd
2025/03/05 13:35:53     | 13:35:53 | apply | Assert Status Conditions | TRY       | DONE  |
2025/03/05 13:35:53 --- PASS: chainsaw (0.00s)
2025/03/05 13:35:53     --- PASS: chainsaw/apply (78.30s)
2025/03/05 13:35:53 PASS
2025/03/05 13:35:53 Tests Summary...
2025/03/05 13:35:53 - Passed  tests 1
2025/03/05 13:35:53 - Failed  tests 0
2025/03/05 13:35:53 - Skipped tests 0
2025/03/05 13:35:53 Done.
2025/03/05 13:35:53 Skipping test 01-update.yaml
2025/03/05 13:35:53 Version: 0.2.0
2025/03/05 13:35:53 Loading default configuration...
2025/03/05 13:35:53 - Using test file: 02-import.yaml
2025/03/05 13:35:53 - TestDirs [/var/folders/ym/wqrhc4wd0mj6b1wbkc25s0_80000gn/T/uptest-e2e/case]
2025/03/05 13:35:53 - SkipDelete true
2025/03/05 13:35:53 - FailFast false
2025/03/05 13:35:53 - ReportFormat ''
2025/03/05 13:35:53 - ReportName 'chainsaw-report'
2025/03/05 13:35:53 - Namespace ''
2025/03/05 13:35:53 - FullName false
2025/03/05 13:35:53 - IncludeTestRegex ''
2025/03/05 13:35:53 - ExcludeTestRegex ''
2025/03/05 13:35:53 - ApplyTimeout 5s
2025/03/05 13:35:53 - AssertTimeout 30s
2025/03/05 13:35:53 - CleanupTimeout 30s
2025/03/05 13:35:53 - DeleteTimeout 15s
2025/03/05 13:35:53 - ErrorTimeout 30s
2025/03/05 13:35:53 - ExecTimeout 5s
2025/03/05 13:35:53 - Parallel 1
2025/03/05 13:35:53 - NoCluster false
2025/03/05 13:35:53 Loading tests...
2025/03/05 13:35:54 - import (/var/folders/ym/wqrhc4wd0mj6b1wbkc25s0_80000gn/T/uptest-e2e/case)
2025/03/05 13:35:54 Loading values...
2025/03/05 13:35:54 Running tests...
2025/03/05 13:35:54 === RUN   chainsaw
2025/03/05 13:35:54 === PAUSE chainsaw
2025/03/05 13:35:54 === CONT  chainsaw
2025/03/05 13:35:54 === RUN   chainsaw/import
2025/03/05 13:35:54 === PAUSE chainsaw/import
2025/03/05 13:35:54 === CONT  chainsaw/import
2025/03/05 13:35:54     | 13:35:54 | import | @setup                           | CREATE    | OK    | v1/Namespace @ chainsaw-assured-caiman
2025/03/05 13:35:54     | 13:35:54 | import | Remove State                     | TRY       | RUN   |
2025/03/05 13:35:54     | 13:35:54 | import | Remove State                     | SCRIPT    | RUN   |
2025/03/05 13:35:54         === COMMAND
2025/03/05 13:35:54         /bin/sh -c ${KUBECTL} annotate xargo.gitops.platform.upbound.io/configuration-gitops-argocd crossplane.io/paused=true --overwrite
2025/03/05 13:35:54         ${KUBECTL} scale deployment crossplane -n ${CROSSPLANE_NAMESPACE} --replicas=0 --timeout 10s
2025/03/05 13:35:54         ${KUBECTL} -n ${CROSSPLANE_NAMESPACE} get deploy --no-headers -o custom-columns=":metadata.name" | grep "provider-" | xargs ${KUBECTL} -n ${CROSSPLANE_NAMESPACE} scale deploy --replicas=0
2025/03/05 13:35:55     | 13:35:55 | import | Remove State                     | SCRIPT    | LOG   |
2025/03/05 13:35:55         === STDOUT
2025/03/05 13:35:55         xargo.gitops.platform.upbound.io/configuration-gitops-argocd annotated
2025/03/05 13:35:55         deployment.apps/crossplane scaled
2025/03/05 13:35:55         deployment.apps/upbound-provider-helm-9fcd9ac09d15 scaled
2025/03/05 13:35:55         deployment.apps/upbound-provider-kubernetes-71953a1e5c15 scaled
2025/03/05 13:35:55     | 13:35:55 | import | Remove State                     | SCRIPT    | DONE  |
2025/03/05 13:35:55     | 13:35:55 | import | Remove State                     | SLEEP     | RUN   |
2025/03/05 13:36:05     | 13:36:05 | import | Remove State                     | SLEEP     | DONE  |
2025/03/05 13:36:05     | 13:36:05 | import | Remove State                     | SCRIPT    | RUN   |
2025/03/05 13:36:05         === COMMAND
2025/03/05 13:36:05         /bin/sh -c ${KUBECTL} scale deployment crossplane -n ${CROSSPLANE_NAMESPACE} --replicas=1 --timeout 10s
2025/03/05 13:36:05         ${KUBECTL} -n ${CROSSPLANE_NAMESPACE} get deploy --no-headers -o custom-columns=":metadata.name" | grep "provider-" | xargs ${KUBECTL} -n ${CROSSPLANE_NAMESPACE} scale deploy --replicas=1
2025/03/05 13:36:05         curl -sL https://raw.githubusercontent.com/crossplane/uptest/main/hack/check_endpoints.sh -o /tmp/check_endpoints.sh && chmod +x /tmp/check_endpoints.sh
2025/03/05 13:36:05         curl -sL https://raw.githubusercontent.com/crossplane/uptest/main/hack/patch.sh -o /tmp/patch.sh && chmod +x /tmp/patch.sh
2025/03/05 13:36:05         /tmp/check_endpoints.sh
2025/03/05 13:36:05         /tmp/patch.sh xargo.gitops.platform.upbound.io configuration-gitops-argocd
2025/03/05 13:36:05         ${KUBECTL} annotate xargo.gitops.platform.upbound.io/configuration-gitops-argocd --all crossplane.io/paused=false --overwrite
2025/03/05 13:36:14     | 13:36:14 | import | Remove State                     | SCRIPT    | LOG   |
2025/03/05 13:36:14         === STDOUT
2025/03/05 13:36:14         deployment.apps/crossplane scaled
2025/03/05 13:36:14         deployment.apps/upbound-provider-helm-9fcd9ac09d15 scaled
2025/03/05 13:36:14         deployment.apps/upbound-provider-kubernetes-71953a1e5c15 scaled
2025/03/05 13:36:14         upbound-provider-helm - No served ports
2025/03/05 13:36:14         upbound-provider-helm - Ports present
2025/03/05 13:36:14         upbound-provider-kubernetes - Ports present
2025/03/05 13:36:14         xargo.gitops.platform.upbound.io/configuration-gitops-argocd patched
2025/03/05 13:36:14         Successfully patched xargo.gitops.platform.upbound.io/configuration-gitops-argocd
2025/03/05 13:36:14         xargo.gitops.platform.upbound.io/configuration-gitops-argocd annotated
2025/03/05 13:36:14         xargo.gitops.platform.upbound.io/configuration-gitops-argocd annotated
2025/03/05 13:36:14         === STDERR
2025/03/05 13:36:14         Retrying... (1/10)
2025/03/05 13:36:14     | 13:36:14 | import | Remove State                     | SCRIPT    | DONE  |
2025/03/05 13:36:14     | 13:36:14 | import | Remove State                     | TRY       | DONE  |
2025/03/05 13:36:14     | 13:36:14 | import | Assert Status Conditions and IDs | TRY       | RUN   |
2025/03/05 13:36:14     | 13:36:14 | import | Assert Status Conditions and IDs | ASSERT    | RUN   | gitops.platform.upbound.io/v1alpha1/XArgo @ configuration-gitops-argocd
2025/03/05 13:36:15     | 13:36:15 | import | Assert Status Conditions and IDs | ASSERT    | DONE  | gitops.platform.upbound.io/v1alpha1/XArgo @ configuration-gitops-argocd
2025/03/05 13:36:15     | 13:36:15 | import | Assert Status Conditions and IDs | ASSERT    | RUN   | gitops.platform.upbound.io/v1alpha1/XArgo @ configuration-gitops-argocd
2025/03/05 13:36:15     | 13:36:15 | import | Assert Status Conditions and IDs | ASSERT    | DONE  | gitops.platform.upbound.io/v1alpha1/XArgo @ configuration-gitops-argocd
2025/03/05 13:36:15     | 13:36:15 | import | Assert Status Conditions and IDs | TRY       | DONE  |
2025/03/05 13:36:15 --- PASS: chainsaw (0.00s)
2025/03/05 13:36:15     --- PASS: chainsaw/import (21.07s)
2025/03/05 13:36:15 PASS
2025/03/05 13:36:15 Tests Summary...
2025/03/05 13:36:15 - Passed  tests 1
2025/03/05 13:36:15 - Failed  tests 0
2025/03/05 13:36:15 - Skipped tests 0
2025/03/05 13:36:15 Done.
2025/03/05 13:36:15 Version: 0.2.0
2025/03/05 13:36:15 Loading default configuration...
2025/03/05 13:36:15 - Using test file: 03-delete.yaml
2025/03/05 13:36:15 - TestDirs [/var/folders/ym/wqrhc4wd0mj6b1wbkc25s0_80000gn/T/uptest-e2e/case]
2025/03/05 13:36:15 - SkipDelete true
2025/03/05 13:36:15 - FailFast false
2025/03/05 13:36:15 - ReportFormat ''
2025/03/05 13:36:15 - ReportName 'chainsaw-report'
2025/03/05 13:36:15 - Namespace ''
2025/03/05 13:36:15 - FullName false
2025/03/05 13:36:15 - IncludeTestRegex ''
2025/03/05 13:36:15 - ExcludeTestRegex ''
2025/03/05 13:36:15 - ApplyTimeout 5s
2025/03/05 13:36:15 - AssertTimeout 30s
2025/03/05 13:36:15 - CleanupTimeout 30s
2025/03/05 13:36:15 - DeleteTimeout 15s
2025/03/05 13:36:15 - ErrorTimeout 30s
2025/03/05 13:36:15 - ExecTimeout 5s
2025/03/05 13:36:15 - Parallel 1
2025/03/05 13:36:15 - NoCluster false
2025/03/05 13:36:15 Loading tests...
2025/03/05 13:36:15 - delete (/var/folders/ym/wqrhc4wd0mj6b1wbkc25s0_80000gn/T/uptest-e2e/case)
2025/03/05 13:36:15 Loading values...
2025/03/05 13:36:15 Running tests...
2025/03/05 13:36:15 === RUN   chainsaw
2025/03/05 13:36:15 === PAUSE chainsaw
2025/03/05 13:36:15 === CONT  chainsaw
2025/03/05 13:36:15 === RUN   chainsaw/delete
2025/03/05 13:36:15 === PAUSE chainsaw/delete
2025/03/05 13:36:15 === CONT  chainsaw/delete
2025/03/05 13:36:15     | 13:36:15 | delete | @setup           | CREATE    | OK    | v1/Namespace @ chainsaw-rested-bison
2025/03/05 13:36:15     | 13:36:15 | delete | Delete Resources | TRY       | RUN   |
2025/03/05 13:36:15     | 13:36:15 | delete | Delete Resources | SCRIPT    | RUN   |
2025/03/05 13:36:15         === COMMAND
2025/03/05 13:36:15         /bin/sh -c ${KUBECTL} delete xargo.gitops.platform.upbound.io/configuration-gitops-argocd --wait=false --ignore-not-found
2025/03/05 13:36:15     | 13:36:15 | delete | Delete Resources | SCRIPT    | LOG   |
2025/03/05 13:36:15         === STDOUT
2025/03/05 13:36:15         xargo.gitops.platform.upbound.io "configuration-gitops-argocd" deleted
2025/03/05 13:36:15     | 13:36:15 | delete | Delete Resources | SCRIPT    | DONE  |
2025/03/05 13:36:15     | 13:36:15 | delete | Delete Resources | TRY       | DONE  |
2025/03/05 13:36:15     | 13:36:15 | delete | Assert Deletion  | TRY       | RUN   |
2025/03/05 13:36:15     | 13:36:15 | delete | Assert Deletion  | CMD       | RUN   |
2025/03/05 13:36:15         === COMMAND
2025/03/05 13:36:15         /usr/local/bin/kubectl wait xargo.v1alpha1.gitops.platform.upbound.io --for=delete configuration-gitops-argocd --timeout 40m0s
2025/03/05 13:36:15     | 13:36:15 | delete | Assert Deletion  | CMD       | DONE  |
2025/03/05 13:36:15     | 13:36:15 | delete | Assert Deletion  | SCRIPT    | RUN   |
2025/03/05 13:36:15         === COMMAND
2025/03/05 13:36:15         /bin/sh -c ${KUBECTL} wait managed --all --for=delete --timeout -1s
2025/03/05 13:36:28     | 13:36:28 | delete | Assert Deletion  | SCRIPT    | LOG   |
2025/03/05 13:36:28         === STDOUT
2025/03/05 13:36:28         release.helm.crossplane.io/configuration-gitops-argocd-hgtw4 condition met
2025/03/05 13:36:28         object.kubernetes.crossplane.io/configuration-gitops-argocd-n9t5f condition met
2025/03/05 13:36:28     | 13:36:28 | delete | Assert Deletion  | SCRIPT    | DONE  |
2025/03/05 13:36:28     | 13:36:28 | delete | Assert Deletion  | TRY       | DONE  |
2025/03/05 13:36:28 --- PASS: chainsaw (0.00s)
2025/03/05 13:36:28     --- PASS: chainsaw/delete (13.36s)
2025/03/05 13:36:28 PASS
2025/03/05 13:36:28 Tests Summary...
2025/03/05 13:36:28 - Passed  tests 1
2025/03/05 13:36:28 - Failed  tests 0
2025/03/05 13:36:28 - Skipped tests 0
2025/03/05 13:36:28 Done.
13:36:28 [ OK ] running automated tests
```
